src/android/SplashScreen.java

```diff
@@ -151,13 +193,40 @@ public Object onMessage(String id, Object data) {
         }
         return null;
     }
+    
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+
+        // Reload splash drawable when orientation changes if so configured.
+        if (newConfig.orientation != orientation) {
+            orientation = newConfig.orientation;
+            reloadDrawable();
+        }
+    }
 
-    private void removeSplashScreen() {
+    private void reloadDrawable() {
+        if (isReloadOnOrientationChange() && splashImageView != null) {
+            final int drawableId = preferences.getInteger("SplashDrawableId", 0);
+            if (drawableId != 0) {
+                cordova.getActivity().runOnUiThread(new Runnable() {
```

Comment:
Originally (in 3.7.x) reload of the drawable was triggered from onMessage and I just followed the pattern I saw in the code of explicitly calling "runOnUiThread". You're probably right that onConfigurationChanged is always called on UI thread. I'll verify and remove the Runnable.